src/components/theme/theme-provider.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import {
+  createTheme,
+  ThemeProvider as MuiThemeProvider,
+} from "@mui/material/styles";
+import { useMemo, PropsWithChildren, useContext, useState } from "react";
+import React from "react";
+
+interface ThemeProviderProps {}
+
+interface ThemeContextProps {
+  mode: "light" | "dark";
+  toggleColorMode: () => void;
+}
+
+const ThemeContext = React.createContext<ThemeContextProps | undefined>(
+  undefined
+);
+
+export const useThemeContext = () => {
+  const context = useContext(ThemeContext);
+  if (!context) {
+    throw new Error("useThemeContext must be used within a ThemeProvider");
+  }
+  return context;
+};
+
+function ThemeProvider(props: PropsWithChildren<ThemeProviderProps>) {
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+
+  const [mode, setMode] = useState<"light" | "dark">(
+    prefersDarkMode ? "dark" : "light"
+  );
+
+  const toggleColorMode = () => {
+    setMode((prevMode) => (prevMode === "light" ? "dark" : "light"));
```

Comment:
We currently store `mode` in `useState` only, but this state will be skipped after the page is reloaded. Did you consider saving `mode` in `localStorage`?